lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -244,7 +244,7 @@ def formatted_message_and_backtrace(colorizer)
         end
 
         def exception_backtrace
-          exception.backtrace || []
+          RSpec.configuration.backtrace_formatter.format_backtrace(exception.backtrace) || []
```

Comment:
The snippet shows the code that triggered the failure, and the backtrace shows the route taken to get there, its not `snippet_exclusion_patterns` after all. Can you provide a real world example of this not working in a nice way? e.g. some library snippet being extracted rather than a spec? This example isn't realistic of something you wouldn't work at the moment